### ISSUE TITLE
Seems to always refresh the cache in test mode

### ISSUE THREAD

**Issue by zedtux** [sstephenson/sprockets#703]
I have a test suite in my private/close sources Rails 4 application, using Cucumber, Capybara and Poltergeist (all with the latest version) which is using javascript all the time:

*features/env.rb*

```ruby
# Use Javascript in all the tests
Capybara.default_driver = :poltergeist
Capybara.javascript_driver = :poltergeist
```

The test suite were running in an acceptable amount of time until I did the upgrade to Rails 4.2. After the upgrade, Poltergeist were failing with a Timeout error. I have opened the issue teampoltergeist/poltergeist#571 but quickly I have identified that updating the Poltergeist `:timeout` parameter 'solved' the issue.

Since this point on, the test suite runs, but the when calling the first `visit` takes a very long time so I have opened the issue rails/rails#18703 on the Rails project and we tried couple of things but nothing worked but I have noticed that the files at `tmp/cache/assets/test/sprockets` were updated at each runs of the test suite.

To summarise my issue, it seems to me (but I could be wrong) that sprockets is always refreshing its cache, even when I'm not touching any assets files and I don't know why.
The issue seems to come from my project and I try to figure out where.

I have uploaded my `features/env.rb` file [here](https://gist.github.com/zedtux/0ad8d8770bf34b183189) and my `test.rb` [here](https://gist.github.com/zedtux/b8a9bafaea80bc1f3e8b).

* In which case sprockets refresh its cache ?
* Is there a way for me to debug this issue (having debug information) ?
* Is there any other file I could provide you in order to help on this issue ?

**Comment by zedtux** [comment]
Any chance to get an answer on this ?

**Comment by josh** [comment]
Sorry, I'm not too familiar with how poltergeist works. Maybe you should try disabling `config.assets.compile = false` and precompiling assets before these tests to simulate a more production build environment. That would probably be quicker.

**Comment by zedtux** [comment]
Thank you @josh, I will work on this but as far as I remember I have tried this and I got the error that `/assets/application.js` can't be found.
Looking around on the Internet I've read that Rails 4 removed the assets generation without the digest. Also I never found how to make the application adding it.

Anyway let me try it and I'll update the issue.

**Comment by josh** [comment]
Good luck @zedtux, but I don't think I can debug any application specific integrations any further. Thanks for asking though.
